custom_model_runner/datarobot_drum/drum/root_predictors/predict_mixin.py

```diff
@@ -368,19 +368,42 @@ def do_predict_unstructured(self, logger=None):
         return response, response_status
 
     def do_chat(self, logger=None):
+        unsupported_chat_message = (
+            "This model's chat interface was called, but chat is not supported."
+        )
+        undefined_chat_message = (
+            "This model's chat interface was called, but chat() is not implemented."
+        )
+        # _predictor is a BaseLanguagePredictor attribute of PredictionServer;
+        # see PredictionServer.__init__()
+        if not self._predictor.supports_chat():
+            if logger is not None:
+                logger.error(unsupported_chat_message)
+            return (
+                {"message": "ERROR: " + unsupported_chat_message},
+                HTTP_422_UNPROCESSABLE_ENTITY,
```

Comment:
I think 404 is more appropriate than 422, because its about interface "not found" than any issue in the input request.

Reply:
I agree, 400/422 is kind of actionable (go fix your request since its a bad request), but this one is not actionable for the caller.